### DISCUSSION TITLE
How do I run SchemaChange with no deployment but with updating the change_history table ?

### DISCUSSION THREAD

**Discussion by zroytman** [Snowflake-Labs/schemachange#381] [Q&A]
** What are you trying to accomplish? **
I'm using the SchemaChange as part of our Azure DevOps pipeline. As we already have many DB objects deployed, I'm looking for the way to load the CHANGE_HISTORY table for the first time without deploying (compiling) the objects in the DB.

** What options have you tried so far ? **
I've tried to run the SchemaChange in Dry Mode and expected to get exactly what I need, means updating the CHANGE_HISTORY table without compiling the DB objects themselves. In real, Dry Mode doesn't compile the objects (which is great) but doesn't update the CHANGE_HISTORY table too (which is bad for me).

So is there any way to run the SchemaChange in the mode that I need? Any additional flag I missed?

Thanks.

**Comment by michaeleyre** [comment]
I have a similar problem.  We have a list of SQL scripts that have already been executed in our development environment and we want to start using schema change, but we want to record the current scripts in the CHANGE_HISTORY table without executing them.

**Comment by kotha** [comment]
Hi,
We ran into similar use case and followed these steps, using some python scripts.

1. Generate DDL's for all the scripts using get_ddl
2. Save the scripts into folders with pre-defined naming convention. We named files in such a way that they are executed in the order we need.
3. Had another python script to scan all the files and generate the checksum (we used the function used in schemachange to make sure the same checksum is generated)
`hashlib.sha224(content.encode('utf-8')).hexdigest()`

Insert the entries generated from the above step into the schema_change_history table

now apply schema change with **dry-run: false** and you should not see script changes getting applied
Thanks

**Comment by zroytman** [comment]
@kotha Thanks. That is exactly what I did ;-)

**Comment by sfc-gh-tmathew** [comment]
I know it is an old item but just to see if this alternative approach is an acceptable option @zroytman @michaeleyre @kotha 

Assuming you have existing scripts that do not have the V, R, A prefixes, can you not start with deployment using the V, R, A format from the cutover date you start using schemachange? 

OR 

if you have a pre-schemachange folder and post-schemachange folder in the repo such that you call schemachange by setting the root-folder to post-schemachange and its contents and when ever you make changes to a file in the pre-schemachange section, move those files into post schemachange folder to be processed by schemachange.

Are these options feasible?

**Reply by zroytman** [reply]
Both the options are not feasible in our case:

**First option**: we don't create new script for each change, but manage the same one. For example, in case we have the procedure named MY_PROC in the db, we create its script just once and name it R__MY_PROC.sql, so each time we need to change the procedure, we change the same script R__MY_PROC.sql - this is the way to manage history/versioning on the procedure. So in order to create the first (existing in the db) version for the procedure, the first creation of the script R__MY_PROC.sql shouldn't be executed - just update the schemachange history table.

**Second option**: managing two folders and moving the changed script from one folder to another means manual managing of changes/versions. We don't want to make it manually - we just update the existing script in the GIT managed folder without moving it as every manual operation/manipulation can be forgotten so we won't know why some change wasn't deployed... Everything should be autimatic.

So the right option is to make schemachange be able to ignore deployment to db - just make update the history table like the object has been deployed.

Thanks.
